numba/compiler.py

```diff
@@ -128,54 +127,65 @@ def __repr__(self):
         return ', '.join(vals)
 
 
-class _StageResult(object):
-    def __init__(self, stage, ok, result=None, exception=None,
-                 early_escape=False):
-        self.stage = stage
-        self.ok = ok
-        self.early_escape = early_escape
-        self._result = result
-        self._exception = exception
-
-    @property
-    def result(self):
-        assert self.ok
-        return self._result
+class _EarlyPipelineCompletion(BaseException):
```

Comment:
This should really be an Exception, not a BaseException. BaseException is only for certain standard exceptions such as SystemExit.